cpp/src/rolling/nth_element.cuh

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/aggregation.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/detail/gather.cuh>
+#include <cudf/detail/iterator.cuh>
+#include <cudf/utilities/bit.hpp>
+
+#include <limits>
+#include <rmm/exec_policy.hpp>
+
+namespace cudf::detail::rolling {
+
+auto constexpr NULL_INDEX = std::numeric_limits<size_type>::min();  // For nullifying with gather.
```

Comment:
```suggestion
auto __device__ constexpr NULL_INDEX = std::numeric_limits<size_type>::min();  // For nullifying with gather.
```

Reply:
Why is this necessary? It's a constexpr, and the compiler is not complaining.

Reply:
This compile flag we use to build libcudf should make this work automatically:
https://github.com/rapidsai/cudf/blob/b2f77fada4d9d352ce5d224bff03ada17aff390c/cpp/cmake/Modules/ConfigureCUDA.cmake#L19
Sounds like a compiler bug found in the referenced PR. I would not add `__device__` here if the compiler does not complain. We should probably investigate and possible open a bug against the compiler for whatever happened in 11019.

Reply:
I just tried to pull this and compile it. This time my compiler doesn't complain. It's weird @davidwendt.

Reply:
I also tried to modify the `constexpr` in the file of the old PR and this time I don't have compiler error! My compiler is still the same, so maybe cudf cmake has been updated somehow to overcome this.

Reply:
I think there was some solar flare activity recently.

Reply:
I remember the PR to which @ttnghia refers. The need for `__device__` at that point was surprising.

Reply:
For completeness, we also should consider this: https://github.com/rapidsai/cudf/issues/7795
And I would also add that we should not assume this compile-flag is used by those compiling with our public headers.

Reply:
IIUC, we don't need `__device__` for `constexpr`. I'm resolving this comment. Please reopen, if I've misunderstood.